src/infra/core/src/repos/dataset_repository_local_fs.rs

```diff
@@ -546,15 +579,27 @@ impl DatasetStorageStrategy for DatasetSingleTenantStorageStrategy {
 
         let dataset_path = self.dataset_path_impl(dataset_alias);
         if !dataset_path.exists() {
+            use tokio_stream::StreamExt;
+
+            let mut all_datasets_stream = self.get_all_datasets();
+
+            while let Some(dataset_handle) = all_datasets_stream.try_next().await.unwrap() {
+                if &dataset_handle.alias == dataset_alias {
+                    return self
+                        .resolve_dataset_handle(
+                            &self.root.join(&dataset_handle.alias.dataset_name),
+                            &dataset_handle.alias,
+                        )
+                        .await;
+                }
+            }
             return Err(ResolveDatasetError::NotFound(DatasetNotFoundError {
                 dataset_ref: dataset_alias.as_local_ref(),
             }));
         }
-        let summary = self
-            .attempt_resolving_summary_via_path(&dataset_path, dataset_alias)
-            .await?;
 
-        Ok(DatasetHandle::new(summary.id, dataset_alias.clone()))
+        self.resolve_dataset_handle(&dataset_path, dataset_alias)
```

Comment:
When we are resolving an Alias to Handle - let's make sure that handle contains an alias in its "true form".

In other words:
- If a dataset was created as `Account/Foo`
- and we call `resolve` by `account/foo`
- let's make sure the handle has alias `Account/Foo`

Also please take a look at `create_dataset` method in this file and imagine a situation where:
- we are creating a new dataset as `Account/Foo`
- repo already contains dataset `account/bar`
- to me it seems like we're missing some step to normalize the `Account` into `account` before creating `account/Foo`

Reply:
Added